applications/StructuralMechanicsApplication/custom_elements/base_solid_element.cpp

```diff
@@ -887,33 +888,28 @@ void BaseSolidElement::CalculateOnIntegrationPoints(
 
             Values.SetStrainVector(this_constitutive_variables.StrainVector);
 
+            const std::function<void(const Vector&, const Vector&, double&)> retrieve_vm_stress_2d =
```

Comment:
why do you need a function here?

a very simple solution is to template the function CaclulateVonMisesEquivalentStress in the InputVectoryTypes and then just call it. this way you can avoid any copying.